### PR TITLE
arch/arm: Remove all lazy fpu related code

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
since it is broken and inefficient, and then removed by:
```
commit dc961baaea17107109d94575e9bb5d9fb725b801
Author: chao.an <anchao@xiaomi.com>
Date:   Thu Apr 14 18:07:14 2022 +0800

    arm/armv7-[a|r]: move fpu save/restore to assembly handler

    Save/Restore FPU registers in C environment is dangerous practive,
    which cannot guarantee the compiler won't generate the assembly code
    with float point registers, especially in interrupt handling

    Signed-off-by: chao.an <anchao@xiaomi.com>

commit 8d66dbc0680cb81f91674ace814b2b8a94c49cd2
Author: chao.an <anchao@xiaomi.com>
Date:   Thu Apr 7 13:48:04 2022 +0800

    arm/armv[7|8]-m: skip the fpu save/restore if stack frame is integer-only

    Signed-off-by: chao.an <anchao@xiaomi.com>
```
please reference #6072, #6001 and #5646 for more info.

## Impact
No, remove the dead code

## Testing
Pass CI

depends on https://github.com/apache/incubator-nuttx/pull/6080.

### CODE REVIEW

File: arch/arm/src/lpc31xx/lpc31_decodeirq.c [hunk 1]
```diff
@@ -99,6 +99,7 @@ uint32_t *arm_decodeirq(uint32_t *regs)
 
           if (regs != CURRENT_REGS)
             {
+#ifdef CONFIG_ARCH_ADDRENV
```
Comment (pkarashchenko) [inline arch/arm/src/lpc31xx/lpc31_decodeirq.c:102]:
```suggestion
```

**Review by pkarashchenko — CHANGES_REQUESTED** [review]

File: arch/arm/src/lpc31xx/lpc31_decodeirq.c (hunk 1 above)
Comment (xiaoxiang781216) [reply arch/arm/src/lpc31xx/lpc31_decodeirq.c]: Done.

**Review by pkarashchenko — APPROVED** [review]

**Comment by pkarashchenko** [comment]
`ineffient` -> `inefficient` in commit message and PR description

File: boards/arm/stm32/stm32f4discovery/configs/wifi/defconfig [hunk 2]
```diff
@@ -20,7 +20,6 @@ CONFIG_ARCH_CHIP_STM32=y
 CONFIG_ARCH_CHIP_STM32F407VG=y
 CONFIG_ARCH_INTERRUPTSTACK=2048
 CONFIG_ARCH_STACKDUMP=y
-CONFIG_ARMV7M_LAZYFPU=y
```
Comment (pkarashchenko) [inline boards/arm/stm32/stm32f4discovery/configs/wifi/defconfig:23]: Probably need to remove `CONFIG_TESTING_OSTEST_FPUSIZE=y` here and in some other configs where `CONFIG_ARMV7M_LAZYFPU=y` is removed.

Comment (xiaoxiang781216) [reply boards/arm/stm32/stm32f4discovery/configs/wifi/defconfig:23]: @anchao will provide a new patch improving fpu test design:

1. Remove CONFIG_TESTING_OSTEST_FPUSIZE config
2. Merge related code from board/chip to arch common

Let's wait his PR.

**Comment by xiaoxiang781216** [comment]
> `ineffient` -> `inefficient` in commit message and PR description

Done.

File: boards/arm/stm32/stm32f4discovery/configs/wifi/defconfig (hunk 2 above)
Comment (xiaoxiang781216) [reply boards/arm/stm32/stm32f4discovery/configs/wifi/defconfig:23]: @anchao 's patch is merged, let's continue this PR.

File: arch/arm/src/common/arm_internal.h [hunk 3]
```diff
@@ -89,67 +89,10 @@
...
-
-#  if defined(CONFIG_ARCH_FPU)
-#    define arm_savestate(regs)  (regs = (FAR uint32_t *)CURRENT_REGS, arm_savefpu(regs))
-#  else
-#    define arm_savestate(regs)  (regs = (FAR uint32_t *)CURRENT_REGS)
-#  endif
-#  define arm_restorestate(regs) (CURRENT_REGS = regs)
-
-#endif
+#define arm_savestate(regs)    (regs = (FAR uint32_t *)CURRENT_REGS)
```
Comment (pkarashchenko) [inline arch/arm/src/common/arm_internal.h:94]:
```suggestion
#define arm_savestate(regs)    (regs = (uint32_t *)CURRENT_REGS)
```

Comment (xiaoxiang781216) [reply arch/arm/src/common/arm_internal.h:157]: Done.

**Review by pkarashchenko — APPROVED** [review]

**Comment by masayuki2009** [comment]
@xiaoxiang781216 

I noticed that fpu_test with spresense:smp failed.

```
user_main: FPU test 
Starting task FPU#1 
FPU#1: pass 1 
fpu_test: Started task FPU#1 at PID=27 
Starting task FPU#2 
FPU#2: pass 1 
fpu_test: Started task FPU#2 at PID=28 
ERROR FPU#1: save1 and save2 do not match 
Values before waiting (save1) (0x2d027204): 
    0000: 2d02c8d0 000000e0 2d027204 2d0272d8 42d11c6e 400601fb 85a238dc 3ffc03f6 
    0008: 00000001 00000001 ffffffe9 40300fda 00000000 f01b866e 400921f9 d3458cd2 
    0010: 401309d9 00000000 00000000 00000000 00000000 00000000 00000000 00000000 
    0018: 00000000 00000000 00000000 00000001 2d027204 80dc3370 101243f2 00000000 
    0020: 0d00c6f1 0d00823c 21000000 00000000 00000000 00000000 00000000 00000000 
    0028: 00000000 00000000 00000000 00000000 00000000 00000000 ERROR FPU#2: save1 and save2 do not match 
00000000 402df84d 
Values before waiting (save1) (0x2d0273cc): 
    0030: 3fe01fb4 40984ecf 40490fd0 00000010 0d00c6bb 
    0000: 2d02d410 000000e0 2d0273cc 2d0274a0 2336023d 400f40e9 
2336023d 400740e9 
Values after waiting (save2) (0x2d0272d8): 
    0008: 00000002 00000001 ffffffe9 407a0749 00000000 f01b866e 0000: 2d02c8d0 000000e0 2d027204 2d0272d8 42d11c6e 400601fb 401921f9 cb535009 
85a238dc 3ffc03f6 
    0010: 401f9ad6 00000000 00000000 00000000 00000000 00000000 0008: 00000001 00000001 ffffffe9 40300fda 00000000 f01b866e 00000000 00000000 
400921f9 d3458cd2 
    0018: 00000000 00000000 00000000 00000001 2d0273cc 80dc3370 0010: 401309d9 00000000 00000000 00000000 00000000 00000000 103243f2 00000000 
00000000 00000000 
    0020: 0d00c6f1 0d00823c 21000000 00000000 00000000 00000000 0018: 00000000 00000000 00000000 00000001 2d0272d8 00000001 00000000 00000000 
00000000 98968000 
    0028: 00000000 00000000 00000000 00000000 00000000 00000000 0020: 0d00c733 0d00823c 61000000 00000000 00000000 00000000 00000000 402df84d 
00000000 00000000 
    0030: 403a0749 40fcd6b7 40c90fd0 00000010 0d00c6bb 
    0028: 00000000 00000000 00000000 00000000 00000000 00000000 
00000000 402df84d 
Values after waiting (save2) (0x2d0274a0): 
    0030: 3fe01fb4 40984ecf 40490fd0 00000010 0d00c72d 
    0000: 2d02d410 000000e0 2d0273cc 2d0274a0 2336023d 400f40e9 
2336023d 400740e9 
    0008: 00000002 00000001 ffffffe9 407a0749 00000000 f01b866e 401921f9 cb535009 
    0010: 401f9ad6 00000000 00000000 00000000 00000000 00000000 00000000 00000000 
    0018: 00000000 00000000 00000000 00000001 2d0274a0 00000001 00000000 98968000 
    0020: 0d00c733 0d00823c 61000000 00000000 00000000 00000000 00000000 00000000 
    0028: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 402df84d 
    0030: 403a0749 40fcd6b7 40c90fd0 00000010 0d00c72d 
```

**Comment by anchao** [comment]
> I noticed that fpu_test with spresense:smp failed.

@masayuki2009 san, I have fix this issue, please review:
https://github.com/apache/incubator-nuttx/pull/6114

**Pull request merged**